app/grandchallenge/components/views.py

```diff
@@ -244,6 +244,52 @@ def new_interface_url(self):
         raise NotImplementedError
 
 
+class CIVSetUpdateViewMixin(CIVSetFormMixin):
+    def process_data_for_object(self, data):
+        """Updates the object"""
+        instance = self.object
+        non_civ_data_keys = self.form_class.Meta.non_civ_fields
+
+        save = False
+        for key in non_civ_data_keys:
+            value = data.get(key, empty)
+            if value is not empty and value != getattr(instance, key):
+                setattr(instance, key, value)
+                save = True
+        if save:
+            instance.save()
+
+        for key in data:
+            if key in non_civ_data_keys:
+                continue
+            instance.create_civ(
+                ci_slug=key, new_value=data[key], user=self.request.user
+            )
+
+        return instance
+
+
+class CIVSetCreateViewMixin(CIVSetFormMixin):
+    def create_object(self, non_civ_data):

```

Comment:
Seems a bit strange to have a special method for object creation here. Cant this be constructed in a general way without the consumer having to override this method?

Reply:
The crux is the base_object reference i.e. 'archive' vs 'reader_study'. I'll see if I can cook something alternatively up, using the model directly and passing the linkage information somewhere.